### PR TITLE
Use f.convert('L') to read grayscale in read_image()

### PR DESCRIPTION
**Pull request by 69guitar1015**
f.convert('L') may be natural way to read grayscale image rather than f.convert('P')

### CODE REVIEW

**Comment by Hakuyume** [comment]
`’L’` sometimes returns wrong value when it is used for label images.
On [CMP facade database](http://cmp.felk.cvut.cz/~tylecr1/facade/), `'L'` returns `[0, 225]` where it only contains 12 classes. `'P'` returns `[0, 11]` properly.

```pyhon
import chainercv
import numpy as np

label = chainercv.utils.read_image(
    'CMP_facade_DB/base/cmp_b0001.png', dtype=np.int32, color=False)[0]
print(label.max())
```

**Comment by 69guitar1015** [comment]
sounds good!

File: chainercv/utils/image/read_label.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import numpy as np
+from PIL import Image
+
+
+def read_label(path, dtype=np.uint8):
+    """Read an label image from a file.
+
+    This function reads an label image from given file. If reading label
```
Comment (Hakuyume) [inline chainercv/utils/image/read_label.py]: `an label` -> `a label`

File: tests/utils_tests/image_tests/test_read_label.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+from chainercv.utils import read_label
+from chainercv.utils import write_image
+
+
+@testing.parameterize(*testing.product({
+    'size': [(48, 32)],
+    'suffix': ['bmp', 'jpg', 'png'],
+    'dtype': [np.float32, np.uint8, bool],
+}))
+class TestReadImage(unittest.TestCase):
```
Comment (Hakuyume) [inline tests/utils_tests/image_tests/test_read_label.py]: `TestReadLabel`

File: chainercv/utils/image/read_label.py (hunk 1 above)
Comment (69guitar1015) [reply chainercv/utils/image/read_label.py]: fixed

File: tests/utils_tests/image_tests/test_read_label.py (hunk 2 above)
Comment (69guitar1015) [reply tests/utils_tests/image_tests/test_read_label.py]: fixed

**Comment by Hakuyume** [comment]
@69guitar1015 Thank you for updating. Could you replace `read_image` with `read_label` where it is used for reading label images (mainly in dataset classes)?

**Comment by 69guitar1015** [comment]
@Hakuyume sorry for forgetting replacement. replacement was done. please check it.

**Comment by Hakuyume** [comment]
How about setting the default `dtype` to `np.int32` like `read_image` with `np.float32`?

File: chainercv/datasets/voc/voc_semantic_segmentation_dataset.py [hunk 3]
```diff
@@ -64,7 +65,7 @@ def _get_label(self, i):
             self.data_dir, 'SegmentationClass', self.ids[i] + '.png')
         # TODO(yuyu2172): Find an option to load properly even with cv2.
         with chainer.using_config('cv_read_image_backend', 'PIL'):
```
Comment (Hakuyume) [inline chainercv/datasets/voc/voc_semantic_segmentation_dataset.py]: Can we remove this comment and `with` statement?

**Comment by Hakuyume** [comment]
@69guitar1015 https://github.com/chainer/chainercv/pull/695/commits/6f09fb646f61abaeef2a64e49b07d4576418e591 this modification breaks flake8 check. Could you fix it? (just remove `import chainer`)

**Comment by 69guitar1015** [comment]
I'm sorry. now fixed

**Pull request merged**